chapter09/index.adoc

```diff
@@ -0,0 +1,451 @@
+
+=== Tracing Concepts (Spans, Traces and Context Propagation)
+
+Tracing is a powerful tool for understanding the flow of requests through a distributed system. Tracing concepts such as *spans*, *traces*, and *context propagation* are fundamental to achieving observability and gaining insights into the system's behavior.
+
+==== Spans
+
+A *span* is the basic unit of work in tracing. It represents a single operation or task performed by a service, such as an HTTP request, a database query, or a computation. Each span contains metadata, including:
+
+* *Operation Name*: Describes the activity (e.g., +HTTP GET /orders+).
+* *Start Time and Duration*: Captures when the operation started and how long it took.
+* *Attributes*: Key-value pairs providing context (e.g., user IDs, resource names, HTTP status codes).
+* *Parent Span ID*: Indicates the parent span, forming a relationship within a trace.
+
+Spans may also include additional data like logs and events, which help provide a detailed view of the operation's lifecycle. Spans are connected together to form a trace, which provides a visual representation of the flow of a request. This can be helpful for identifying bottlenecks and performance issues.
+
+==== Traces
+
+A *trace* is a collection of related spans that represent the end-to-end execution of a request or transaction. It provides a holistic view of how a single request flows through the system, including interactions between services. Traces often form a tree structure, where the root span represents the entry point (e.g., a user request), and child spans represent subsequent operations. +
+For example:
+
+* A user requests order details.
+* The API gateway records the root span.
+* Child spans are created as the request is forwarded to the order service, which interacts with a database.
+
+==== Context Propagation
+
+*Context propagation* ensures that trace metadata, such as trace IDs and span IDs, travels with requests as they move through different services and threads. This metadata allows spans created in different services to be correlated into a single trace. Context propagation is vital for connecting distributed spans and understanding their relationships. +
+
+In Java, context propagation is often implemented using thread-local variables or libraries like MicroProfile Context Propagation. This ensures that tracing information is maintained seamlessly across thread pools, asynchronous tasks, and service boundaries.
+
+Together, these concepts form the foundation of distributed tracing, enabling developers to monitor, analyze, and optimize the performance of their microservices effectively.
+
+=== Instrumenting MicroProfile Telemetry
+
+Instrumenting Open Telemetry in an E-commerce Application
```

Comment:
I think this should be

```suggestion
=== Instrumenting with MicroProfile Telemetry

Instrumenting an E-commerce Application with MicroProfile Telemetry
```

as we're talking about using MicroProfile Telemetry to instrument an application, not instrumenting MicroProfile Telemetry itself.